src/Subscriber/PaymentSubscriber.php

```diff
@@ -215,15 +223,18 @@ public function onContextTokenUpdate(SalesChannelContextSwitchEvent $event)
 
         // Save state data, only if Adyen payment method is selected
         if ($event->getRequestDataBag()->get('adyenStateData')) {
+            if (!$event->getRequestDataBag()->has('paymentMethodId')) {
```

Comment:
maybe add an error similar to this to the `else` part of the condition `if ($paymentMethod->getPluginId() === $this->adyenPluginId) {` ? because then adyen payment data was provided for a non adyen payment method.